ndk-build/src/cargo.rs

```diff
@@ -14,12 +14,43 @@ pub fn cargo_ndk(
     let clang_target = format!("--target={}{}", target.ndk_llvm_triple(), sdk_version);
     let mut cargo = Command::new("cargo");
 
-    // Read initial RUSTFLAGS
+    const SEP: &str = "\x1f";
+
+    // Read initial CARGO_ENCODED_/RUSTFLAGS
     let mut rustflags = match std::env::var("CARGO_ENCODED_RUSTFLAGS") {
-        Ok(val) => val,
-        Err(std::env::VarError::NotPresent) => "".to_string(),
+        Ok(val) => {
+            if std::env::var_os("RUSTFLAGS").is_some() {
+                panic!(
+                    "Both `CARGO_ENCODED_RUSTFLAGS` and `RUSTFLAGS` were found in the environment, please clear one or the other before invoking this script"
+                );
+            }
+
+            val
+        }
+        Err(std::env::VarError::NotPresent) => {
+            match std::env::var("RUSTFLAGS") {
+                Ok(val) => {
+                    // Note cargo also does this very simplified parsing of only
+                    // splitting on spaces, ignoring things like quoting
```

Comment:
> ignoring things like quoting

As clearly stated in the documentation, and implied by my suggested improved commit/PR description (I'll apply that while merging if you don't beat me to it) this is by design / specification. This comment only causes confusion, implying that some form of escaping by quoting (and maybe other "_things_"?) is available.

Again a request to reference https://doc.rust-lang.org/cargo/reference/environment-variables.html, clearly quoting:

> This string is split by whitespace; for a more robust encoding of multiple arguments, see CARGO_ENCODED_RUSTFLAGS.